mistralrs-core/src/gguf/gguf_tokenizer.rs

```diff
@@ -140,15 +179,49 @@ fn unigram_tokenizer(p: &PropsGGUF) -> Result<(Tokenizer, TokenizerKind, Vec<Str
         .build()?;
 
     // Add special tokens (bos, eos, unk):
-    let mut special_tokens = Vec::<String>::new();
-    for token_id in [bos, eos, unk] {
-        let token = p.tokens[token_id as usize].as_str();
+    let special_tokens = add_special_tokens(p, &mut tokenizer, bos, eos, Some(unk));
 
-        special_tokens.push(token.to_owned());
-        tokenizer.add_special_tokens(&[AddedToken::from(token.to_owned(), true)]);
+    Ok((tokenizer, TokenizerKind::Unigram, special_tokens))
+}
+
+fn bpe_tokenizer(p: &PropsGGUF) -> Result<(Tokenizer, TokenizerKind, AddedTokensCollection)> {
+    let merges = p
```

Comment:
Some context to add if you think it might be helpful for those like myself that aren't as familiar but wanted to understand what was going on here 😅 

```suggestion
    // BPE merges have each string item as a space-delimited pair:
    // https://github.com/EricLBuehler/mistral.rs/pull/397#discussion_r1631988370
    let merges = p
```

Reply:
Yes, thank you!